internal/v1/server.go

```diff
@@ -469,12 +476,25 @@ func (h *Handlers) ComposeImage(ctx echo.Context) error {
 		return err
 	}
 
-	rawCR, err := json.Marshal(composeRequest)
+	dbComposeRequest := DBComposeRequest{
+		Customizations: composeRequest.Customizations,
+		Distribution:   composeRequest.Distribution,
+		ImageRequests:  composeRequest.ImageRequests,
+	}
+
+	rawCR, err := json.Marshal(dbComposeRequest)
 	if err != nil {
 		return err
 	}
 
-	err = h.server.db.InsertCompose(composeResult.Id, idHeader.Identity.AccountNumber, idHeader.Identity.Internal.OrgId, rawCR)
+	var imageName string
+	if composeRequest.ImageName == nil {
+		imageName = ""
```

Comment:
default value of  a string is `""` so you can simplify to
```
if composeRequest.ImageName != nil {
     imageName = *....
}
```

Reply:
Don't derefence it, if it's non-required in the response, it won't be specified.

And I think the UI can then just check, if 'name' is in the response -> use the name, otherwise fallback to the uuid

Reply:
I wouldn't ever dereference. Just store the ComposeEntry with a pointer directly.